### PR TITLE
chore: update react-query

### PR DESCRIPTION
**Pull request by yomybaby**
### What changed?
- Replaced various instances of `useTanQuery` with `useSuspenseTanQuery` in the codebase.
- Adjusted imports to ensure `useSuspenseTanQuery` is used from the `@tanstack/react-query` package.
- Adjusted suspense-related settings in query hooks to ensure proper behavior.

### Why make this change?
- This change ensures that suspense behavior is consistently applied where necessary, optimizing the React components' data fetching mechanism with suspense.
- This is particularly useful for ensuring that components display fallback content while data is being fetched, improving the user experience.

### How to test?
1. Verify that the application builds correctly without errors related to the query hooks.
2. Check that components using `useSuspenseTanQuery` correctly handle data fetching with suspense, displaying fallback content where appropriate.
3. Ensure the application behaves as expected without regressions in functionality.

### CODE REVIEW

**Comment by yomybaby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2591?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2591** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2591?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2590** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2590?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/ResourceGroupSelect.tsx [hunk 1]
```diff
@@ -80,6 +80,7 @@ const ResourceGroupSelect: React.FC<ResourceGroupSelectProps> = ({
       ]);
     },
     staleTime: 0,
+    fetchKey: fetchKey,
   });
```
Comment (yomybaby) [inline react/src/components/ResourceGroupSelect.tsx:79]: This is a use case for `fetchKey` instead of updating `queryKey`.

**Review by agatha197 — CHANGES_REQUESTED** [review]
I got this error when creating a new session. Can you check it?

**Review by agatha197 — CHANGES_REQUESTED** [review]
Please resolve the conflicts.

**Comment by yomybaby** [comment]
I resolve that error. Please check again.

**Review by ironAiken2 — APPROVED** [review]
LGTM!

(I tested for no build errors, good handling with suspense fallback, and refetching on fetchkey updates).

**Pull request merged**